tests/test_connector.py

```diff
@@ -1195,16 +1201,18 @@ def capture_future(*args, features_future=None, **kwargs):
             base.mock_calls,
             [
                 unittest.mock.call.metadata.ssl_context_factory(),
+                unittest.mock.call.metadata.
+                    ssl_context_factory().set_alpn_protos([b"xmpp-client"]),
```

Comment:
Do you have objections to changing this testcase to test that _context_factory_factory is invoked and used instead of testing that it does something similar to what the context_factory created by the _context_factory_factory does?

Reply:
And test what `_context_factory_factory(...)(...)` does separately? I agree that this would be a cleaner separation of tests. I am a bit reluctant because the change might introduce bugs in the test and thereby mask a regression due to the change in connector.py (note, how the test did not change apart from the added call to `set_alpn_protos`! So it demonstrates that the behaviour before and after the change is the same with respect to the mocked objects.).

Reply:
Okay, the changes where minimal enough so that I am comfortable with the change (I could simply move the call-sequence assertion to the new test). And once again I have to remark: `unittest.mock` is pure joy ;).